client/entities/global.go

```diff
@@ -0,0 +1,198 @@
+package entities
+
+import (
+	"encoding/json"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"github.com/contiv/volplugin/client"
+	"github.com/contiv/volplugin/errors"
+)
+
+const (
+	// DefaultGlobalTTL is the TTL used when no TTL exists.
+	DefaultGlobalTTL = 30 * time.Second
+	// DefaultTimeout is the standard command timeout when none is provided.
+	DefaultTimeout = 10 * time.Minute
+
+	timeoutFixBase   = time.Minute
+	ttlFixBase       = time.Second
+	defaultMountPath = "/mnt/ceph"
+
+	globalPathName = "global-config"
+)
+
+// NewGlobalFromJSON transforms json into a global. Conforms to
+// client.Entity interface.
+func NewGlobalFromJSON(content []byte) (*Global, error) {
+	global := NewGlobal()
+
+	if err := json.Unmarshal(content, global); err != nil {
+		return nil, err
+	}
+
+	return global.setEmpty(), nil
+}
+
+// NewGlobal returns global config with preset defaults. Conforms to
+// client.Entity interface.
+func NewGlobal() *Global {
+	return &Global{
+		TTL:       DefaultGlobalTTL,
+		MountPath: defaultMountPath,
+		Timeout:   DefaultTimeout,
+	}
+}
+
+// Path returns the path of the global configuration
+func (g *Global) Path(p *client.Pather) (string, error) {
+	inter, err := p.Append(globalPathName)
+	if err != nil {
+		return "", err
+	}
+
+	return inter.String(), nil
+}
+
+// Payload returns the global configuration in JSON form.
+func (g *Global) Payload() ([]byte, error) {
+	content, err := json.Marshal(g)
+	if err != nil {
+		return nil, errors.MarshalGlobal.Combine(err)
+	}
+
+	return content, nil
+}
+
+// Published returns a copy of the current global with the parameters adjusted
+// to fit the published representation. To see the internal/system/canonical
+// version, please see Canonical() below.
+//
+// It is very important that you do not run this function multiple times
+// against the same data set. It will adjust the parameters twice.
+func (g *Global) Published() *Global {
```

Comment:
What would be an example usage of this function?  I don't see it being called anywhere internally